src/cockpit/packages.py

```diff
@@ -194,11 +194,37 @@ def __init__(self, value: JsonObject):
 
 
 class Condition:
-    def __init__(self, value: JsonObject):
-        try:
-            (self.name, self.value), = value.items()
-        except ValueError as exc:
-            raise JsonError(value, 'must contain exactly one key/value pair') from exc
+    def get_condition_files(self) -> Iterable[str]:
+        raise NotImplementedError
+
+    def check(self, path_exists: Callable[[str], bool]) -> bool:
+        raise NotImplementedError
+
+
+class UnknownCondition(Condition):
+    def __init__(self, value: JsonObject, path: Path) -> None:
+        self.value = value
+        self.path = path
+
+    def get_condition_files(self) -> Iterable[str]:
+        return []
+
+    def check(self, path_exists: Callable[[str], bool]) -> bool:
+        # do *not* ignore manifests with unknown predicates, for forward compatibility
+        logger.warning('  %s: ignoring unknown predicate in manifest: %s', self.path, self.value)
```

Comment:
I'd have done the warning from inside of `parse_condition()` (ie: when we actually encounter the unknown condition) and then called this `TrivialCondition` or something which just always returns true without further warning, but this is serious deckchair shifting at this point.  It's *theoretically* possible that we will fail to see a message with a file that has an unknown predicate if it's inside of an any for which one of the earlier conditions already matched, since then we'll never hit this .check()... but .... at some point you probably stop caring :)